### PR TITLE
add pharo-6.1 to lineup

### PR DESCRIPTION
**Pull request by dalehenrich**

### CODE REVIEW

**Comment by dalehenrich** [comment]
Looks like [same 6 tests fail in Pharo6.0](https://travis-ci.org/Metacello/metacello/jobs/272071045)

**Comment by estebanlm** [comment]
@dalehenrich 
I have no idea why Pharo 5.0 tests are failing, and I cannot reproduce it in my machine :(
any idea?

**Comment by krono** [comment]
Well, 
```
Smalltalk 
 		at: #SessionManager
 		ifPresent: [ :c | c default registerNetworkClassNamed: self name ]
 		ifAbsent:[ Smalltalk addToStartUpList: self ].
```
broke all builds...

**Comment by dalehenrich** [comment]
@estebanlm I agree with @krono it certainly looks like [#at:ifPresent:ifAbsent:](https://travis-ci.org/Metacello/metacello/jobs/280491532#L662) is the culprit ... perhaps in your image, **MCGitBasedNetworkRepository** is already initialized, since that is where the failure appears to be happening on travis?

**Comment by krono** [comment]
Or Pharo has `SmalltalkImage>> #at:ifPresent:ifAbsent:`

**Comment by krono** [comment]
Also, it _would_ work with `Smalltalk globals  #at:ifPresent:ifAbsent:` in Squeak..

**Comment by estebanlm** [comment]
hmmm... ok, I will submit a fix tomorrow.

**Comment by estebanlm** [comment]
and now I screwed the Pharo platform code. 
this thing of monticello branches is annoying.

**Comment by krono** [comment]
What about 
```Smalltalk
Smalltalk globals
		at: #SessionManager
		ifPresent: [ :c | c default registerNetworkClassNamed: self name ]
		ifAbsent:[ Smalltalk addToStartUpList: self ].
```

**Comment by estebanlm** [comment]
meh… maybe… I took #classNamed: because that’s what we have for FFI-Kernel.
but in the merge I lost the baseline… grrr… anyway, retrying :)

**Comment by estebanlm** [comment]
still failing, now for other reasons :(
I will check what I broke tomorrow.

**Comment by estebanlm** [comment]
So. Now, I don't know why Gemstone is failing :)
(also, the cache problem is becoming... annoying)

File: repository/Metacello-MC.package/MetacelloMCProjectSpec.class/instance/loadVersion..st [hunk 1]
```diff
@@ -1,43 +1,44 @@
...
+				loader: mcLoader ]) asSet.
+	MetacelloPlatform current
+		useStackCacheDuring: [ :dict | 
+			| projectCache cachedList |
+			projectCache := dict
+				at: self projectClass
+				ifAbsent: [ dict at: self projectClass put: Dictionary new ].
+			(cachedList := projectCache at: vrsn ifAbsent: [  ]) == nil
+				ifTrue: [ projectCache at: vrsn put: list ]
+				ifFalse: [ (cachedList size = list size and: [ cachedList includesAll: list ])
```
Comment (dalehenrich) [inline repository/Metacello-MC.package/MetacelloMCProjectSpec.class/instance/loadVersion..st]: #includesAll: is [breaking the GemStone build](https://travis-ci.org/Metacello/metacello/jobs/282157190#L9322) and [the Squeak builds](https://travis-ci.org/Metacello/metacello/jobs/282157199#L8628) ... need to use portable methods or we have to move more methods to Metacello platform ...

**Comment by estebanlm** [comment]
yes I saw that... but I did not make that change, that's why is weird.

**Comment by dalehenrich** [comment]
I've commented on your checking ... #includesAll: is not portabel ... #includesAllOf: is portable ... or at least used to be ...

**Comment by estebanlm** [comment]
mmm... #includesAllOf: was victim of autodeprecation tool and I didn't see it :( 
yet *another* issue to fix :'(

**Comment by dalehenrich** [comment]
With regards to the "cache problem", I assume you are referring to [this](https://travis-ci.org/Metacello/metacello/jobs/282157197#L586)? I consider clearing the cache a workaround to a "pharo problem"

Have you looked into who is doing the `mv`? This seems to be a bug in the smalltalkCI pharo download code somewhere ...

**Comment by dalehenrich** [comment]
Yeah, writing portable code is real pain when one or more platforms decide to deprecate formerly portable methods ... it's why ANSI Smalltalk was created ... it =should be possible for a Smalltalk programmer to write portable code with the expectation that the lifetime of that portable code is greater then 1 year ... Perhaps a new effort at redefining what should be considered as portable should be started?

**Comment by estebanlm** [comment]
yeah... annoying. 
But do not blame me! While I can agree with changes that improves the platform, I dislike the name changes "just because"... (in this case it was because there were two equivalent methods and who chose which one to keep didn't know about compatibility :( )

**Comment by dalehenrich** [comment]
I'm not blaming, but I am suggesting that a stake be put into the ground for a set of protocol that can be depended upon to not change between platforms and for the foreseeable future :)

**Comment by estebanlm** [comment]
ok, cache problem aside, seems that this time I nailed it (finally!)

**Comment by dalehenrich** [comment]
I've cleared the caches and restarted the pharo6.1 build for the PR and the build on your the source branch ... they should both be green before merging a pull request ...

**Comment by dalehenrich** [comment]
@estebanlm this pupy dog is ready to go ... you want to merge the pull request and exercise your owner super powers?

**Comment by estebanlm** [comment]
ok :)

**Pull request merged**